internal/services/instance_types_service.go

```diff
@@ -55,13 +56,14 @@ func ListInstanceTypes(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	res, err := newEC2Client.ListInstanceTypes()
+	res, err := newEC2Client.ListInstanceTypesWithPaginator()
 	if err != nil {
 		renderError(w, r, payloads.NewAWSError(r.Context(), "can't list EC2 instance types", err))
 		return
 	}
 
 	instances, err := ec2.NewInstanceTypes(r.Context(), res)
+	ctxval.Logger(r.Context()).Debug().Msgf("Number of AWS EC2 instance types after filtering: %d", len(*instances))
```

Comment:
You have the same message in the function as well, so pick a place and log only once please. I would suggest a trace level as it's very detailed information. Also I suggest to introduce a `logger` variable here as someone else could copy this code in this function and we would end up calling `ctxval.Logger` multiple times (which is pretty expensive operation). The compiler will probably optimize single-use variable anyway automatically so there is nothing to worry about.

Reply:
it is the number of instances **after filtering** ( filter unsupported instance types and create new ones if they support more than one arch)

Reply:
Right, could you then just do a single trace here, count it before, after and make a single message containing both information? Let me explain why I worry about: ending up with massive amount of log messages typically lead to decreasing verbosity and that could be harmful for the future, so I think it is worth trying our best to keep the amount of information as small as possible.